### ISSUE TITLE
Unexpected reload when accessing '/' route in html5mode

### ISSUE THREAD

**Issue by avechina** [angular/angular.js#13271]
There is a issue with the html5mode when we first load the application.

If you go to http://angularjs.learnwebtutorials.com/demo/24/, you will find a very simple html5mode application.

If you navigate to http://angularjs.learnwebtutorials.com/demo/24/#!/content you will see that angular redirects you to **/demo/24/content** and loads the correct template.

However, if you navigate to http://angularjs.learnwebtutorials.com/demo/24/#!/, angular will redirect you to **/demo/24/** but right before loading the root template, the browser will trigger a new request to the same page loading the application for the second time.

I am having this problem with angular 1.4.2 and 1.4.6.

**Comment by Narretz** [comment]
Yeah, that looks odd. For further reference, you can cause the bug like this in the demo:
1. Go to http://angularjs.learnwebtutorials.com/demo/24/content
2. Modify the url to http://angularjs.learnwebtutorials.com/demo/24/#!/ (with or without closing slash)
